src/main/java/com/checkout/payments/apm/KlarnaSource.java

```diff
@@ -0,0 +1,85 @@
+package com.checkout.payments.apm;
+
+import com.checkout.apm.klarna.KlarnaProduct;
+import com.checkout.common.CountryCode;
+import com.checkout.common.PaymentSource;
+import com.checkout.payments.RequestSource;
+import com.google.gson.annotations.SerializedName;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+
+@Data
+@Builder
+public final class KlarnaSource implements RequestSource {
+
+    private static final PaymentSource type = PaymentSource.KLARNA;
+
+    @SerializedName("authorization_token")
+    private final String authorizationToken;
+
+    private final String locale;
+
+    @SerializedName("purchase_country")
+    private final CountryCode purchaseCountry;
+
+    @SerializedName("tax_amount")
+    private final Integer taxAmount;
+
+    private final Customer customer;
+
+    private final List<KlarnaProduct> products;
+
+    @Data
+    @Builder
+    public static class BillingAddress {
+
+        @SerializedName("given_name")
+        private String givenName;
+
+        @SerializedName("family_name")
+        private String familyName;
+
+        private String email;
+
+        private String title;
+
+        @SerializedName("street_address")
+        private String streetAddress;
+
+        @SerializedName("street_address2")
+        private String streetAddress2;
+
+        @SerializedName("postal_code")
+        private String postalCode;
+
+        private String city;
+
+        private String phone;
```

Comment:
Please double-check is this is a String (usually is the class `Phone`).

Reply:
Yes it is only a String https://docs.checkout.com/payments/payment-methods/invoice-and-pay-later/klarna#Klarna-Placetheorderplaceorder